doc/design/s3-undelete.md

```diff
@@ -0,0 +1,69 @@
+# **S3 Undelete**
+
+## **Why is “undelete” necessary?**
+
+The core value of the DANDI Archive comes from the data we host. The process for getting this data into DANDI often involves coordination between several people to get an extremely large volume of data annotated with useful metadata and uploaded to our system. Because of the amount of time and work involved in this process, we need to minimize the risk of accidental data loss to the greatest extent that is possible and reasonable. Additionally, we would like to implement “garbage collection” in the future, which involves programmatically clearing out stale asset blobs from S3. All of this leads to a desire to be able to “undelete” an s3 object that has been deleted.
+
+Our ultimate goal is to prevent data loss from application programming errors. With protection such as an undelete capability, we will be safer in implementing application features that involve intentional deletion of data. Any bugs we introduce while doing so are far less likely to destroy data that was not supposed to be deleted.
+
+The original GitHub issue around this feature request can be found at [https://github.com/dandi/dandi-archive/issues/524](https://github.com/dandi/dandi-archive/issues/524). Although the issue asks for a Deep Glacier storage tier, the design in this document solves the underlying problem differently (and in a more robust way). Below we address the possible usage of a Deep Glacier tiered bucket as a solution to the orthogonal problem of data ******backup******, which addresses a different problem than the undelete capability described in this document.
```

Comment:
why here 6 `*`s for bold, instead of e.g. just 2?  below you also used 4 `*`s -- any particular reason for extra emphasis?

Note, although we I think have not introduced such convention to design docs but we use it in other projects already: It would be nice to start using "one sentence a line" paragraph formatting.  In this case it would have helped to break down a possible patch to the paragraph (now 3 long sentences) for the formatting into a much more manageable unit - a sentence.

Reply:
Ah that's weird, I used a markdown editor and then exported it to a `.md` file. I guess I didn't look closely enough at the markdown it generated, I'll normalize these.